src/constructs/core/parameters.ts

```diff
@@ -86,3 +86,29 @@ export class GuAmiParameter extends GuParameter {
     });
   }
 }
+
+export const arnRegex = "arn:aws:[a-z0-9]*:[a-z0-9\\-]*:[0-9]{12}:.*";
+
+export class GuArnParameter extends GuStringParameter {
+  constructor(scope: GuStack, id: string, props: GuNoTypeParameterProps) {
+    super(scope, id, {
+      ...props,
+      allowedPattern: arnRegex,
+      constraintDescription: "Must be a valid ARN, eg: arn:partition:service:region:account-id:resource-id",
+    });
+  }
+}
+
+const s3BucketRegex = "(?!^(\\d{1,3}\\.){3}\\d{1,3}$)(^[a-z0-9]([a-z0-9-]*(\\.[a-z0-9])?)*$(?<!\\-))";
+export const s3ArnRegex = `arn:aws:s3:::${s3BucketRegex}*`;
+
+export class GuS3ArnParameter extends GuStringParameter {
```

Comment:
Looking at the tests, should we rename this to `GuS3ObjectArnParameter` as it looks like the regex is for an object path and not a bucket name?

Reply:
Good shout, and done!